openhands/sdk/agent/base.py

```diff
@@ -67,6 +69,54 @@ def env_context(self) -> EnvContext | None:
         """Returns the environment context used by the Agent."""
         return self._env_context
 
+    @property
```

Comment:
@openhands I think we might want to save these to conversation state as we want the Agent to be as stateless as possible (i.e. not maintain its own state, so we only need to serialize the conversation for the agent to work)


finally redo the example in examples/confirmation_mode_example.py so that we have a small example of a conversation with confirmation mode